at_secondary/at_persistence_secondary_server/lib/src/keystore/hive_keystore.dart

```diff
@@ -235,9 +236,17 @@ class HiveKeystore implements SecondaryKeyStore<String, AtData?, AtMetaData?> {
   Future<int?> remove(String key) async {
     int? result;
     try {
+      bool isKeyPresent =
+          persistenceManager!.getBox().containsKey(Utf7.encode(key));
       await persistenceManager!.getBox().delete(Utf7.encode(key));
-      _metaDataCache.remove(key);
-      result = await _commitLog.commit(key, CommitOp.DELETE);
+      final atMetaData = _removeKeyFromMetadataCache(key);
+      if (atMetaData != null || isKeyPresent) {
+        // add entry to commit log only if key is removed from cache or key is present in keystore
+        result = await _commitLog.commit(key, CommitOp.DELETE);
+      } else {
+        logger.finer(
+            'entry for $key is not present in cache. Skipping commit log delete op');
+      }
       return result;
```

Comment:
what's the impact of returning null (when we skip the commit log delete op) ?

Reply:
remove is called in multiple places. out of which result is consumed in two places
1) local_secondary.delete - data:null will be returned if key is not present in cache/keystore
2) in at_secondary_impl --> _removeMalformedKeys()..we are passing the commitId to  await _commitLog.commitLogKeyStore.remove(commitId); This needs a null check before remove is called.

Reply:
Follow-up questions:
1. What is the impact of local_secondary.delete returning "data:null" ?
2. Has that null check been implemented?

Reply:
An existing functional test fail due to this change
https://github.com/atsign-foundation/at_server/actions/runs/3265641359/jobs/5368172492
When we delete non existent key should we return data:null or KeyNotFoundException?  will bring this up in architecture call.

Reply:
Usual follow-up question: What is the impact of returning data:null vs throwing KeyNotFoundException?

Reply:
I think it would be good to describe the behaviours through the various call chains before this PR, and what they are now. We're trying to implement the scenario where where a key *does* exist, but HiveKeyStore is going to return null when deleting the object, because we are not generating an entry in the commit log.

We've got multiple intertwined things happening here (1) what we do for a keyStore delete if the key does not exist (2) whether or not we generate a commitID for a given key deletion,  (3) what gets returned at the atProtcocol level when this is a server operation while handling a `delete:` verb request (4) what gets returned by LocalSecondary when this is a client-side delete

Reply:
![Screenshot from 2022-10-26 11-28-19](https://user-images.githubusercontent.com/18326522/197946525-543ad5f4-bd1c-4b25-b3b7-88b64cab6b2d.png)